packages/frontend/core/src/desktop/pages/workspace/trash-page.tsx

```diff
@@ -35,23 +39,80 @@ const TrashHeader = () => {
 };
 
 export const TrashPage = () => {
+  const t = useI18n();
+  const collectionRulesService = useService(CollectionRulesService);
   const globalContextService = useService(GlobalContextService);
-  const currentWorkspace = useService(WorkspaceService).workspace;
   const permissionService = useService(WorkspacePermissionService);
+
+  const { restoreFromTrash } = useBlockSuiteMetaHelper();
+  const isActiveView = useIsActiveView();
+
+  const [explorerContextValue] = useState(() =>
+    createDocExplorerContext({
+      displayProperties: [
+        'system:createdAt',
+        'system:updatedAt',
+        'system:tags',
+      ],
+      showMoreOperation: false,
+      showDragHandle: true,
+      showDocPreview: false,
+      quickFavorite: false,
+    })
+  );
+
   const isAdmin = useLiveData(permissionService.permission.isAdmin$);
   const isOwner = useLiveData(permissionService.permission.isOwner$);
-  const docCollection = currentWorkspace.docCollection;
-  const docsService = useService(DocsService);
-  const allTrashPageIds = useLiveData(
-    LiveData.from(docsService.allTrashDocIds$(), [])
+  const groups = useLiveData(explorerContextValue.groups$);
+  const isEmpty =
+    groups.length === 0 ||
+    (groups.length > 0 && groups.every(group => !group.items?.length));
+
+  const handleMultiRestore = useCallback(
+    (ids: string[]) => {
+      ids.forEach(id => {
+        restoreFromTrash(id);
+      });
+      toast(
+        t['com.affine.toastMessage.restored']({
+          title: ids.length > 1 ? 'docs' : 'doc',
+        })
+      );
+    },
+    [restoreFromTrash, t]
   );
 
-  const pageMetas = useBlockSuiteDocMeta(docCollection);
-  const filteredPageMetas = useMemo(() => {
-    return pageMetas.filter(page => allTrashPageIds.includes(page.id));
-  }, [pageMetas, allTrashPageIds]);
+  useEffect(() => {
+    const subscription = collectionRulesService
+      .watch({
+        filters: [
+          {
+            type: 'system',
+            key: 'trash',
+            method: 'is',
+            value: 'true',
+          },
+        ],
+        orderBy: {
+          type: 'system',
+          key: 'updatedAt',
+          desc: true,
+        },
+      })
+      .subscribe(result => {
+        explorerContextValue.groups$.next(result.groups);
+      });
 
-  const isActiveView = useIsActiveView();
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [collectionRulesService, explorerContextValue.groups$]);
+
+  useEffect(() => {
```

Comment:
The way of handling permissions here is not correct. We should check the permissions on each doc separately. I will deal with it in another PR later. you can delete here first.